docs/notebooks/soft_cosine_tutorial.ipynb

```diff
@@ -0,0 +1,613 @@
+    "* [Wicked Spoon](http://en.yelp.be/biz/wicked-spoon-las-vegas).\n",
+    "* [Serendipity 3](http://en.yelp.be/biz/serendipity-3-las-vegas).\n",
+    "* [Bacchanal Buffet](http://en.yelp.be/biz/bacchanal-buffet-las-vegas-7).\n",
+    "* [The Buffet](http://en.yelp.be/biz/the-buffet-las-vegas-6).\n",
+    "\n",
+    "The restaurants we chose were those with the highest number of reviews in the Yelp dataset. Incidentally, they all are on the Las Vegas Boulevard. The corpus we trained Word2Vec on has 27028 documents (reviews), and the corpus we used for `SoftCosineSimilarity` has 6978 documents.\n",
+    "\n",
+    "Below a JSON file with Yelp reviews is read line by line, the text is extracted, tokenized, and stopwords and punctuation are removed.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[nltk_data] Downloading package punkt to /home/witiko/nltk_data...\n",
+      "[nltk_data]   Package punkt is already up-to-date!\n",
+      "[nltk_data] Downloading package stopwords to /home/witiko/nltk_data...\n",
+      "[nltk_data]   Package stopwords is already up-to-date!\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Pre-processing a document.\n",
+    "from nltk.corpus import stopwords\n",
+    "from nltk import download, word_tokenize\n",
+    "download('punkt')  # Download data for tokenizer.\n",
+    "download('stopwords')  # Download stopwords list.\n",
+    "stop_words = stopwords.words('english')\n",
+    "\n",
+    "def preprocess(doc):\n",
+    "    doc = doc.lower()  # Lower the text.\n",
+    "    doc = word_tokenize(doc)  # Split into words.\n",
+    "    doc = [w for w in doc if not w in stop_words]  # Remove stopwords.\n",
+    "    doc = [w for w in doc if w.isalpha()]  # Remove numbers and punctuation.\n",
+    "    return doc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cell took 103.94 seconds to run.\n"
+     ]
+    }
+   ],
+   "source": [
+    "start = time()\n",
+    "\n",
+    "import json\n",
+    "\n",
+    "# Business IDs of the restaurants.\n",
+    "ids = ['4JNXUYY8wbaaDmk3BPzlWw', # Mon Ami Gabi\n",
+    "       'Ffhe2cmRyloz3CCdRGvHtA', # Earl of Sandwich\n",
+    "       'K7lWdNUhCbcnEvI0NhGewg', # Wicked Spoon\n",
+    "       'eoHdUeQDNgQ6WYEnP2aiRw', # Serendipity 3\n",
+    "       'RESDUcs7fIiihp38-d6_6g', # Bacchanal Buffet\n",
+    "       '2weQS-RnoOBhb1KsHKyoSQ'] # The Buffet\n",
+    "\n",
+    "w2v_corpus = []  # Documents to train word2vec on (all 6 restaurants).\n",
+    "scs_corpus = []  # Documents to run queries against (only one restaurant).\n",
+    "documents = []  # scs_corpus, with no pre-processing (so we can see the original documents).\n",
+    "with open('/data/review.json') as data_file:\n",
```

Comment:
I don't think we can, the [license](https://s3-media2.fl.yelpcdn.com/assets/srv0/engineering_pages/e926cc12796d/assets/vendor/yelp-dataset-license.pdf) seems to explicitly forbid any distribution of the dataset (see section C). The dataset is also used in the Word Mover's Distance notebook, but from the description, it contained different data back when the notebook was created, so it seems the dataset is continuously changing.

Reply:
Should I update both the WMD and SCS notebooks to use some open dataset available in `gensim-data`?

Reply:
@Witiko for this case, I see two possible solutions:
- Leave as is (because Yelp dataset is really good for this task, but we can't store it in gensim-data), probably add link, when user can download this dataset
- Choice other datasets (that will be really nice for demonstration + possible to add it to gensim-data by license). @Witiko have you good candidates for it?

Reply:
I use the [SemEval 2015–2017 Task 3 Subtask B](http://alt.qcri.org/semeval2017/task3/) question answering datasets in my thesis. These are also a good fit for the semantic similarity task and are permissively licensed, so it should be possible to add them to `gensim-data`.

Reply:
Nice, please replace Yelp dataset with SemEval + create an issue in `gensim-data` with needed info (guide: https://github.com/RaRe-Technologies/gensim-data#want-to-add-a-new-corpus-or-model), probably add other "tasks" from semeval will be really good too.

Reply:
I am running the evaluations. If all looks fine, I will submit a PR to `gensim-data`. Until then, please consider the current state of the Jupyter Notebook temporary.